src/cartridges/int_adyen_SFRA/cartridge/adyen/utils/adyenHelper.js

```diff
@@ -537,20 +563,53 @@ let adyenHelperObj = {
     return paymentRequest;
   },
 
+  setPaymentInstrumentFields(paymentInstrument, stateData) {
+	if (!(paymentInstrument instanceof dw.order.OrderPaymentInstrument)) {
+		return null;
+	}
+    const paymentMethodType = stateData.paymentMethod.type;
+	// Currently this doesn't set the fields for cards and giftcards, they are handled by other flow
+    if (
+      [constants.PAYMENTMETHODS.SCHEME, constants.PAYMENTMETHODS.GIFTCARD].indexOf(paymentMethodType) ===
+      -1
+    ) {
+        paymentInstrument.custom.adyenPaymentMethod =
+          adyenHelperObj.getAdyenComponentType(stateData.paymentMethod.type);
+        paymentInstrument.custom[
+          `${constants.OMS_NAMESPACE}__Adyen_Payment_Method`
+        ] = adyenHelperObj.getAdyenComponentType(stateData.paymentMethod.type);
+        paymentInstrument.custom.Adyen_Payment_Method_Variant =
+          stateData.paymentMethod.type.toLowerCase();
+        paymentInstrument.custom[
+          `${constants.OMS_NAMESPACE}__Adyen_Payment_Method_Variant`
+        ] = stateData.paymentMethod.type.toLowerCase();
+    }
+  },
+
   // creates a request object to send to the Adyen Checkout API
-  createAdyenRequestObject(orderNo, orderToken, paymentInstrument, customerEmail) {
+  createAdyenRequestObject(
+    orderNo,
+    orderToken,
+    paymentInstrument,
+    customerEmail,
+  ) {
     const jsonObject = JSON.parse(paymentInstrument.custom.adyenPaymentData);
 
     const filteredJson = adyenHelperObj.validateStateData(jsonObject);
     const { stateData } = filteredJson;
 
+    Transaction.wrap(() => {
+      adyenHelperObj.setPaymentInstrumentFields(paymentInstrument, stateData);
```

Comment:
Could we move setPaymentInstrumentFields function call before or after `createAdyenRequestObject` function is called. The reason is `createAdyenRequestObject ` being a helper function should have only one responsibility of returning a request object. To avoid any side effects it would be good to move `setPaymentInstrumentFields` function call a level up.

Reply:
The reasons of using `createAdyenRequestObject()` for setting these fields are:
- We use the validated state data which are also passed to the /payments call
- We have access to `paymentInstrument`

Moving it one step before it's not possible, since the state data will not be present, while moving it after leads to the same issue we had; having translated fields as the data is retrieved from the form.

The state data should return the payment method data in a formatted and generic way, however, in case that doesn't happen, we can use a try/catch block in `setPaymentInstrumentFields()` and throw an error log in case of any inconsistency.